source/common/upstream/upstream_impl.cc

```diff
@@ -170,14 +170,22 @@ void HostSetImpl::updateHosts(HostVectorConstSharedPtr hosts,
   hosts_per_locality_ = std::move(hosts_per_locality);
   healthy_hosts_per_locality_ = std::move(healthy_hosts_per_locality);
   locality_weights_ = std::move(locality_weights);
-  // Rebuild the locality scheduler.
+  // Rebuild the locality scheduler by computing the effective weight of each
+  // locality in this priority. No scheduler is built if we don't have locality weights
+  // (i.e. not using EDS) or when there are 0 healthy hosts in this priority.
+  //
+  // Zero healthy hosts can only happen if the entire cluster is unhealthy, as this HostSet
```

Comment:
Is this confusing selecting a host set during scheduling with building the host set at EDS update time?

Reply:
ah yeah you're right. ill reword